cordova-plugin-outline/apple/src/OutlinePlugin.swift

```diff
@@ -295,4 +299,59 @@ class OutlinePlugin: CDVPlugin {
     }
     return callbackId
   }
+
+  // Migrates local storage files from UIWebView to WKWebView.
+  private func migrateLocalStorage() {
+    let appLibraryDir = FileManager.default.urls(for: .libraryDirectory, in: .userDomainMask)[0]
+
+    var uiWebViewLocalStorage: URL
+    if FileManager.default.fileExists(atPath: appLibraryDir.appendingPathComponent("WebKit/LocalStorage/file__0.localstorage").relativePath) {
+        uiWebViewLocalStorage = appLibraryDir.appendingPathComponent("WebKit/LocalStorage")
+    } else {
+        uiWebViewLocalStorage = appLibraryDir.appendingPathComponent("Caches")
+    }
+
+    // Local storage backing files have the following naming format: $scheme_$hostname_$port.localstorage
+    // With UIWebView, the app used the file:// scheme with no hostname and any port
+    uiWebViewLocalStorage = uiWebViewLocalStorage.appendingPathComponent("file__0.localstorage")
+
+    // With WKWebView, the app uses the app:// scheme with localhost as a hostname and any port.
+    let wkWebViewLocalStorage = appLibraryDir.appendingPathComponent("WebKit/WebsiteData/LocalStorage/app_localhost_0.localstorage")
+
+    // Only copy the local storage files if they don't exist for WKWebView
+    if FileManager.default.fileExists(atPath: wkWebViewLocalStorage.relativePath) {
```

Comment:
Why do you need the relative path?

Reply:
`wkWebViewLocalStorage` is a `URL`, `relativePath` returns a string representation of the path.